src/chapters/auth/request-filtering.rst

```diff
@@ -0,0 +1,204 @@
+
+      // Don't require sign-in for whitelisted pages
+      if (isWhitelisted(request.getRequestURI())) {
+         // returning true indicates that the request may proceed
+         return true;
+      }
+
+      HttpSession session = request.getSession();
+      User user = authenticationController.getUserFromSession(session);
+
+      // The user is logged in
+      if (user != null) {
+         return true;
+      }
+
+      // The user is NOT logged in
+      response.sendRedirect("/login");
+      return false;
+   }
+
+``request.getRequestURI()`` returns the request path (see `the docs <https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpServletRequest.html>`_ for more details). Lines 43-47 check the path against the whitelist, returning true (that is, allowing the request to proceed) if the path is whitelisted. 
+
+With our filter complete, we simply need to let Spring know about it to complete our authentication code.
+
+Registering the Filter With Spring
+----------------------------------
+
+.. index:: ! code-based configuration
+
+Spring Boot uses **code-based configuration**, which means that many application configurations can be done using Java classes. We'll use code-based configuration to register our filter.
+
+At the top-level package of the app, ``org.launchcode.codingevents``, create a class named ``WebApplicationConfig`` with the code below:
+
+.. sourcecode:: java
+   :lineno-start: 11
+
+   @Configuration
+   public class WebApplicationConfig implements WebMvcConfigurer {
+
+      // Create spring-managed object to allow the app to access our filter
+      @Bean
+      public AuthenticationFilter authenticationFilter() {
+         return new AuthenticationFilter();
+      }
+
+      // Register the filter with the Spring container
+      @Override
+      public void addInterceptors(InterceptorRegistry registry) {
+         registry.addInterceptor( authenticationFilter() );
+      }
+
+   }
+
+The ``@Configuration`` annotation flags this class to Spring as one that contains configuration code. By implementing ``WebMvcConfigurer``, we ensure that Spring will call our ``addInterceptors`` method during startup, giving us the chance to register our filter. 
+
+The first method---``authenticationFilter``, which is annotated with ``@Bean``---makes our filter available as a Spring-managed class. 
+
+The code in ``WebApplicationConfig`` is hard to understand, and contains some advanced concepts that you haven't fully learned. That's okay. We don't expect you to understand every detail of this class. However, you do need to understand its role in enabling authentication, which is to register our filter with Spring.
+
+Check Your Understanding
+------------------------
+
+.. admonition:: Question
+
+   True/False: Request filtering takes place before any controller is called.
+
+   #. True
+   #. False
+
+.. ans: True
+
+.. admonition:: Question
+
+   True/False: Paths included in a whitelist must be exact.
```

Comment:
The way this is phrased is a little confusing. And we should specify that this is how *our* whitelist code works (since there's not standard behavior here). How about: "When our code checks a path against entries in the whitelist, it must match exactly in order for it be accessed without logging in."